### PR TITLE
Add a maven profile & a tool in order to publish to Apache dist

### PR DESCRIPTION
**Pull request by aduprat**

### CODE REVIEW

File: release/publishJars.sh [hunk 1]
```diff
@@ -0,0 +1,107 @@
...
+   echo "    JAMES_SOURCES_PATH: path of the James code"
+   echo "    JAMES_DIST_PATH: path of the James dist svn repo"
+   echo "    RELEASE: release version ex.: 3.3.0"
+   echo "    SHA1: sha1 corresponding of the git commit of the release"
+   echo "    KEY: key footprint to use for signing"
+   exit 1
+}
+
+addAndSign() {
+    sha1sum $1 > $1.sha1
```
Comment (rouazana) [inline release/publishJars.sh]: still using sha1? it's not deprecated?

File: release/publishJars.sh [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+    svn add $1
+    svn propset svn:mime-type application/octet-stream $1
+    svn add $1.sha1
+    svn add $1.asc
+    svn propset svn:mime-type application/octet-stream $1.asc
+}
+
+guiceProduct() {
+    mkdir -p $1/target/$2-$RELEASE
```
Comment (rouazana) [inline release/publishJars.sh]: I bet you did not try with $1 containing spaces ;)

File: release/publishJars.sh [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    cp $1/target/$2.jar $1/target/$2-$RELEASE
+    cp -r $1/target/$2.lib $1/target/$2-$RELEASE
+    cp $JAMES_SOURCES_PATH/server/container/cli/target/james-server-cli.jar $1/target/$2-$RELEASE
+    cp -r $JAMES_SOURCES_PATH/server/container/cli/target/james-server-cli.lib $1/target/$2-$RELEASE
+    cp -r $3 $1/target/$2-$RELEASE
+    cp $1/README.md $1/target/$2-$RELEASE
+    cd $1/target
+    zip -r $2-$RELEASE.zip $2-$RELEASE
+    sha1sum $2-$RELEASE.zip > $2-$RELEASE.zip.sha1
```
Comment (rouazana) [inline release/publishJars.sh]: sha1 again?

File: release/publishJars.sh [hunk 4]
```diff
@@ -0,0 +1,107 @@
...
+    gpg -u $KEY --output $2-$RELEASE.zip.asc --detach-sig $2-$RELEASE.zip
+    cp  $2-$RELEASE.zip $2-$RELEASE.zip.asc $2-$RELEASE.zip.sha1 $JAMES_DIST_PATH/server/$RELEASE
+    svn add $JAMES_DIST_PATH/server/$RELEASE/$2-$RELEASE.zip
+    svn propset svn:mime-type application/octet-stream $JAMES_DIST_PATH/server/$RELEASE/$2-$RELEASE.zip
+    svn add $JAMES_DIST_PATH/server/$RELEASE/$2-$RELEASE.zip.sha1
+    svn add $JAMES_DIST_PATH/server/$RELEASE/$2-$RELEASE.zip.asc
+    svn propset svn:mime-type application/octet-stream $JAMES_DIST_PATH/server/$RELEASE/$2-$RELEASE.zip.asc
```
Comment (rouazana) [inline release/publishJars.sh]: I can only hope your are enough confident not having forgotten any file :)

File: release/publishJars.sh [hunk 5]
```diff
@@ -0,0 +1,107 @@
...
+}
+
+if [ "$#" -ne 5 ]; then
+    printUsage
+fi
+
+JAMES_SOURCES_PATH=$1
```
Comment (rouazana) [inline release/publishJars.sh]: same comment about spaces

**Comment by Arsnael** [comment]
Read it...

**Comment by aduprat** [comment]
Was :green_apple: before CI upgrade

File: release/publishJars.sh (hunk 2 above)
Comment (aduprat) [reply release/publishJars.sh]: C'mon, it's a committer tool; they should be aware of that.

File: release/publishJars.sh (hunk 5 above)
Comment (aduprat) [reply release/publishJars.sh]: same answer

File: release/publishJars.sh (hunk 2 above)
Comment (rouazana) [reply release/publishJars.sh]: aware of what? That you are not allowed to work in a /home/me/My Dev/ directory?

**Comment by aduprat** [comment]
Was :green_apple: before CI restart.

**Comment by aduprat** [comment]
test this please

**Comment by chibenwa** [comment]
Many merge conflict, inactive for very long, opened on linagora repo and not the ASF one...

Reopen on ASF repository if relevant.

**Pull request closed without merging**